mindsdb/integrations/handlers/pinecone_handler/pinecone_handler.py

```diff
@@ -0,0 +1,331 @@
+                {"created_at": {"$gt": "2019-01-01"}}
+            ]
+        }
+        """
+        # we ignore all non-metadata conditions
+        if conditions is None:
+            return None
+        metadata_conditions = [
+            condition
+            for condition in conditions
+            if condition.column.startswith(TableField.METADATA.value)
+        ]
+        if len(metadata_conditions) == 0:
+            return None
+
+        # we translate each metadata condition into a dict
+        pinecone_conditions = []
+        for condition in metadata_conditions:
+            metadata_key = condition.column.split(".")[-1]
+            pinecone_conditions.append(
+                {
+                    metadata_key: {
+                        self._get_pinecone_operator(condition.op): condition.value
+                    }
+                }
+            )
+
+        # we combine all metadata conditions into a single dict
+        metadata_condition = (
+            {"$and": pinecone_conditions}
+            if len(pinecone_conditions) > 1
+            else pinecone_conditions[0]
+        )
+        return metadata_condition
+
+    def _matches_to_dicts(self, matches: List):
+        """Converts the custom pinecone response type to a list of python dict"""
+        return [match.to_dict() for match in matches]
+
+    def connect(self):
+        """Connect to a pinecone database."""
+        try:
+            pinecone.init(api_key=self._client_config["api_key"], environment=self._client_config["environment"])
+            pinecone.list_indexes()
+            self.is_connected = True
+        except Exception as e:
+            log.logger.error(f"Error connecting to Pinecone client, {e}!")
+            self.is_connected = False
+
+    def disconnect(self):
+        """Close the pinecone connection."""
+        if self.is_connected is False:
+            return
+        pinecone.init(api_key="", environment="")
+        self.is_connected = False
+
+    def check_connection(self):
+        """Check the connection to pinecone."""
+        response_code = StatusResponse(False)
+        try:
+            pinecone.list_indexes()
+            response_code.success = True
+        except Exception as e:
+            log.logger.error(f"Error connecting to pinecone , {e}!")
+            response_code.error_message = str(e)
+        return response_code
+
+    def get_tables(self) -> HandlerResponse:
+        """Get the list of indexes in the pinecone database."""
+        indexes = pinecone.list_indexes()
+        indexes_names = pd.DataFrame(
+            columns=["index_name"],
+            data=[index for index in indexes],
+        )
+        return Response(resp_type=RESPONSE_TYPE.TABLE, data_frame=indexes_names)
+
+    def insert(self, table_name: str, data: pd.DataFrame, columns: List[str] = None) -> HandlerResponse:
+        """Insert data into pinecone index passed in through `table_name` parameter."""
+        upsert_size = 99
```

Comment:
Is this parameter important in practice? If so, we may wanna provide users the option to specify it somehow.

Reply:
Oh okay it is the chunk size. Yup perhaps it would be great to store this when creating the engine if the user passes the size through `USING`.

Reply:
This upsert size is something recommended by API, I don't think it is wise to leave this up to the user. I have added a comment there. If you still feel like we should let user decide, I can change it. For now I am marking this resolved. Thank you.